cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -192,8 +189,15 @@ public void dropAllInNamespace(final NamespaceId namespace) throws Exception {
       }
     });
 
-    for (final Location location : locations) {
-      doDrop(namespace.stream(StreamUtils.getStreamNameFromLocation(location)), location);
+    Map<Location, StreamId> locationStreamIdMap = new HashMap<>();
+    for (Location location : locations) {
```

Comment:
I think it is only required to check the visibility here, we can just enforce ADMIN on the stream since we will anyway check the ADMIN privilege on all streams. If it passes, we drop it, if not, we just skip the deletion. And if we delete all the streams, drop the state table. Otherwise, we keep it. I have made the change. Please take a look